hydra/extra/pytest_plugin.py

```diff
@@ -42,6 +44,7 @@ def _(
         config_name: Optional[str],
         overrides: Optional[List[str]],
         strict: Optional[bool] = None,
+        temp_dir: Optional[Path] = None,
```

Comment:
I am tempted to start [requiring named parameters](https://stackoverflow.com/questions/2965271/forced-naming-of-parameters-in-python
) in places like this.

It's a breaking change initially but it will make it much easier to remove things later, for example strict will go away here in the next major version.